### PR TITLE
feat(module:icon): add `provideNzIcon` and `provideNzIconPatch` API

### PR DESCRIPTION
**Pull request by Laffery**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/NG-ZORRO/ng-zorro-antd/blob/master/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [x] Documentation content changes
- [ ] Application (the showcase website) / infrastructure changes
- [ ] Other... Please describe:

## What is the current behavior?

Issue Number: N/A

## What is the new behavior?

Developer can use `provideNzIcon` API in root and `provideNzIconPatch` in lazy-loading module or standalone component to provide icons now.
Of course the former `NzIconModule.forRoot` and `NzIconModule.forChild` API are still supported but not recommended in standalone mode any more.

[USAGE]

```typescript
import { provideNzIcon } from 'ng-zorro-antd/icon';

export const appConfig = {
  providers: [provideNzIcon(icons)]
}
```

```typescript
import { NzIconModule, provideNzIconPatch } from 'ng-zorro-antd/icon';

// in xxx.component.ts
@Component({
  standalone: true,
  imports: [NzIconModule],
  providers: [provideNzIconPatch([QuestionOutline])]
})
class ChildComponent {}

// or in xxx.routes.ts
const routes: Routes = [{
  path: '',
  providers: [provideNzIconPatch([QuestionOutline])],
}]
```

## Does this PR introduce a breaking change?
- [ ] Yes
- [x] No

### CODE REVIEW

File: components/icon/provide-icon.ts [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+ */
+export const provideNzIconPatch = (icons: IconDefinition[]): Provider[] => {
+  return [
+    NzIconPatchService,
+    {
+      provide: NZ_ICONS_PATCH,
+      useValue: icons
+    }
+  ];
+};
```
Comment (HyperLife1119) [inline components/icon/provide-icon.ts:38]: NzIconModule 应该可以直接复用这两个函数：

```ts
export class NzIconModule {
  static forRoot(icons: IconDefinition[]): ModuleWithProviders<NzIconModule> {
    return {
      ngModule: NzIconModule,
      providers: [
        provideNzIcon(icons)
      ]
    };
  }

  static forChild(icons: IconDefinition[]): ModuleWithProviders<NzIconModule> {
    return {
      ngModule: NzIconModule,
      providers: [
        provideNzIconPatch(icons)
      ]
    };
  }
}
```

Comment (HyperLife1119) [reply components/icon/provide-icon.ts:38]: 在之前的 ng g 原理图中，我们还手动定义了一个 provideNzIcons 函数，此时我们需要将它替换为当前的实现。

QQ_1721875061104

Comment (HyperLife1119) [reply components/icon/provide-icon.ts:38]: NIT: 这两个函数的命名需要考虑改为复数形式吗？

provideNzIcons / provideNzIconsPatch

**Review by HyperLife1119 — CHANGES_REQUESTED** [review]

File: components/icon/provide-icon.ts (hunk 1 above)
Comment (Laffery) [reply components/icon/provide-icon.ts:38]: > 在之前的 ng g 原理图中，我们还手动定义了一个 provideNzIcons 函数，此时我们需要将它替换为当前的实现。

赞同，我稍后会更新下 schematics。

命名我们 follow 之前定义的  provideNzIcons 吧，语义上也更合理

**Review by HyperLife1119 — APPROVED** [review]
LGTM

**Pull request merged**